modeshape-jcr/src/main/java/org/modeshape/jcr/cache/document/WritableSessionCache.java

```diff
@@ -948,6 +869,22 @@ public void save( Set<NodeKey> toBeSaved,
     }
 
     /**
+     * Rolling back given transaction caused by given cause.
+     * 
+     * @param txn the transaction 
+     * @param cause roll back cause
+     * @throws Exception roll back cause.
+     */
+    private void rollback(Transaction txn, Exception cause) throws Exception {
+        try {
+            txn.rollback();
+        } catch (Exception e) {
+        } finally {
```

Comment:
that's fine, but in some cases the fact that we cannot rollback a transaction might provide some useful information as to why that is. That's why logging instead of ignoring this exception might be of more use.